rad/src/lib.rs

```diff
@@ -26,56 +28,97 @@ pub mod types;
 
 pub type Result<T> = std::result::Result<T, RadError>;
 
-pub async fn try_data_request(request: &RADRequest) -> RadonReport<RadonTypes> {
+/// The return type of any method executing the entire life cycle of a data request.
+#[derive(Debug, Serialize)]
+pub struct RADRequestExecutionReport {
+    /// Vector of reports about retrieval of data sources.
+    pub retrieve: Vec<RadonReport<RadonTypes>>,
+    /// Report about aggregation of data sources.
+    pub aggregate: RadonReport<RadonTypes>,
+    /// Report about aggregation of reports (reveals, actually).
+    pub tally: RadonReport<RadonTypes>,
+}
+
+pub async fn try_data_request(
+    request: &RADRequest,
+    settings: RadonScriptExecutionSettings,
+    inputs_injection: Option<&[&str]>,
+) -> RADRequestExecutionReport {
     let context = &mut ReportContext::default();
 
-    let retrieve_responses_fut = request
-        .retrieve
-        .iter()
-        .map(|retrieve| run_retrieval_report(retrieve));
+    let mut retrieve_responses = Vec::new();
 
-    let retrieve_responses: Vec<RadonTypes> = futures::future::join_all(retrieve_responses_fut)
-        .await
+    if let Some(inputs) = inputs_injection {
+        for (retrieve, input) in request.retrieve.iter().zip(inputs.iter()) {
```

Comment:
Using `zip` has a problem, it will stop on the shortest iterator. This may introduce hard to debug errors. Since this branch is mainly used for tests, I vote for adding a simple assert:

```suggestion
        assert_eq!(request.retrieve.len(), inputs.len());
        for (retrieve, input) in request.retrieve.iter().zip(inputs.iter()) {
```

And in the documentation of `try_data_request` add a short note about `inputs_injection`: it's used to avoid running the HTTP queries, it must be the same length as the number of sources, etc.